### DISCUSSION TITLE
Should Octokit consider keeping track of package 📦 size to avoid size regressions?

### DISCUSSION THREAD

**Discussion by oscard0m** [octokit/octokit.js#2288] [Ideas]
I got inspired by this issue: https://github.com/octokit/oauth-app.js/issues/330

I think it could be cool to keep track of how much size our packages are occupying and prevent, at the PR stage, if the changes are growing our package unexpectedly.

**Comment by gr2m** [comment]
I think we used to have a bundlesize check, I think we used https://github.com/apps/bundlesize, but it's not for `node_modules` size. I think we used something else for that in the past, or maybe it was in another project ... 🤷🏼 all this to say, I think it's a good idea, as long as it's low maintenance.

**Reply by oscard0m** [reply]
I found this GitHub Action: [Size Limit Action](https://github.com/marketplace/actions/size-limit-action) which is based on [size-limit](https://github.com/ai/size-limit).

It requires us to set the size limit we want. In case the changes made exceed the limit, the PR will need to update the limit in `package.json` as well. 

If you are happy with this approach we can give it a try if you want.

**Reply by gr2m** [reply]
sounds like a good approach. However we don't have bundled files, we would need to do that before the Size Limit Action? If it's really easy, let's try, but if we feel it adds too much complexity, I'd rather not.
